dev/wlp-gradle/subprojects/assemble.gradle

```diff
@@ -24,39 +24,68 @@ task publishWLPJars(type: Copy) {
         // next to the base version of the jar, *except* for jars that are directly
         // referenced in a tool script's classpath...
-        if (((bnd('IBM-Interim-Fixes', '').empty) && (bnd('IBM-Test-Fixes', '').empty))
-                || (project.name.equals("com.ibm.ws.kernel.boot"))
-                || (project.name.equals("com.ibm.ws.kernel.boot.archive"))
-                || (project.name.equals("com.ibm.ws.appclient.boot"))
-                || (project.name.equals("com.ibm.ws.kernel.boot.cmdline"))) {
-            rename '(.*).jar', '$1' + "_${bnd.bFullVersion}.jar"
+        if (((!bnd('IBM-Interim-Fixes', '').empty) || (!bnd('IBM-Test-Fixes', '').empty))
+                && (!project.name.equals("com.ibm.ws.kernel.boot"))
+                && (!project.name.equals("com.ibm.ws.kernel.boot.archive"))
+                && (!project.name.equals("com.ibm.ws.appclient.boot"))
+                && (!project.name.equals("com.ibm.ws.kernel.boot.cmdline"))) {
+            hasIFIXHeaders.put(outerSymbolicName, true)
         }
-        else {
-            rename '(.*).jar', '$1' + "_${bnd.bFullVersion}.${qualifier}.jar"
-        }
-    }
-    if (!bnd('-sub', '').empty) {
-        def hasIFIXHeaders = [:]
-        fileTree(dir: projectDir, include: bnd('-sub', '')).each { subBndFile ->
-            Properties subBndProperties = new Properties()
-            subBndFile.withInputStream { subBndProperties.load(it) }
-            if (subBndProperties.getProperty("IBM-Interim-Fixes") != null
-                    || subBndProperties.getProperty("IBM-Test-Fixes") != null) {
-                hasIFIXHeaders.put(subBndProperties.getProperty("Bundle-SymbolicName"), true)
+
+        if (!bnd('-sub', '').empty) {
+            fileTree(dir: projectDir, include: bnd('-sub', '')).each { subBndFile ->
+                Properties subBndProperties = new Properties()
+                subBndFile.withInputStream { subBndProperties.load(it) }
+                def symbolicName = subBndProperties.getProperty("Bundle-SymbolicName")
+                if (symbolicName != null) {
+                    int index = symbolicName.indexOf(";")
+                    if (index != -1) {
+                        symbolicName = symbolicName.substring(0, index).trim()
+                    }
+
+                    if (subBndProperties.getProperty("IBM-Interim-Fixes") != null
+                        || subBndProperties.getProperty("IBM-Test-Fixes") != null) {
+                        hasIFIXHeaders.put(symbolicName, true)
+                    }
+
+                    def subBVersion = subBndProperties.getProperty('bVersion')
+                    if (outerVersion.equals(subBVersion)) {
+                        fullVersions.put(symbolicName, outerFullVersion)
+                    } else {
+                        fullVersions.put(symbolicName, subBVersion + outerFullVersion.substring(outerVersion.length()))
+                    }
+                }
```

Comment:
Comments to explain what this is doing and why it's needed would be helpful.

Possibly some of it could be refactored into separate methods to make it more readable.

Reply:
I will add comments.  Refactoring into separate methods may not be as feasible.  It is mainly a swizzle of existing logic though.

Reply:
Comments have been added